core/templates/pages/exploration-editor-page/statistics-tab/statistics-tab.directive.ts

```diff
@@ -60,61 +58,79 @@ angular.module('oppia').directive('statisticsTab', [
         'DateTimeFormatService', 'ExplorationDataService',
         'ExplorationFeaturesService', 'ExplorationStatesService',
         'ReadOnlyExplorationBackendApiService', 'RouterService',
-        'StateImprovementSuggestionService',
-        'StateInteractionStatsService', 'StatesObjectFactory',
-        'UrlInterpolationService', 'IMPROVE_TYPE_INCOMPLETE',
+        'StateImprovementSuggestionService', 'StateInteractionStatsService',
+        'StatesObjectFactory', 'UrlInterpolationService',
+        'IMPROVE_TYPE_INCOMPLETE',
         function(
             $http, $scope, $uibModal, AlertsService, ComputeGraphService,
             DateTimeFormatService, ExplorationDataService,
             ExplorationFeaturesService, ExplorationStatesService,
             ReadOnlyExplorationBackendApiService, RouterService,
-            StateImprovementSuggestionService,
-            StateInteractionStatsService, StatesObjectFactory,
-            UrlInterpolationService, IMPROVE_TYPE_INCOMPLETE) {
-          var ctrl = this;
-          var _EXPLORATION_STATS_VERSION_ALL = 'all';
-          var stateStatsModalIsOpen = false;
+            StateImprovementSuggestionService, StateInteractionStatsService,
+            StatesObjectFactory, UrlInterpolationService,
+            IMPROVE_TYPE_INCOMPLETE) {
+          const ctrl = this;
+          const newCompletionRatePieChartOptions = title => ({
+            chartAreaWidth: 240,
+            colors: ['#d8d8d8', '#008808', 'blue'],
+            height: 270,
+            left: 20,
+            legendPosition: 'right',
+            pieHole: 0.6,
+            pieSliceBorderColor: 'black',
+            pieSliceTextStyleColor: 'black',
+            title: title,
+            width: 240,
+          });
 
-          ctrl.getLocaleAbbreviatedDatetimeString = function(millisSinceEpoch) {
-            return DateTimeFormatService.getLocaleAbbreviatedDatetimeString(
-              millisSinceEpoch);
+          const loadLatestExploration = () => {
+            return ReadOnlyExplorationBackendApiService.loadLatestExploration(
+              ExplorationDataService.explorationId
+            ).then(response => {
+              const { init_state_name: initStateName, states } = (
```

Comment:
Can we do an explicit conversion to a domain object? I'm not comfortable with using this destructuring syntax since it apparently does not throw any errors if the keys provided are incorrect.

Reply:
Obsolete, I've kept this logic the same.